evap/evaluation/models.py

```diff
@@ -312,16 +316,20 @@ def midterm_grade_documents(self):
     def responsibles_names(self):
         return ", ".join([responsible.full_name for responsible in self.responsibles.all().order_by("last_name")])
 
+    @property
+    def all_evaluations_finished(self):
+        return all(evaluation.state in ['evaluationFinished', 'reviewed', 'published'] for evaluation in self.evaluations.all())
```

Comment:
Have you thought about writing this so we ask the database whether this is true or false? Something like `not self.evaluations.exclude(-allowed_states-).exists()`? Guess it would be faster.